pkg/apis/v1alpha1/stage_types.go

```diff
@@ -74,7 +74,13 @@ type StageSpec struct {
 	// Next indicates that this stage will be moved to.
 	Next StageNext `json:"next"`
 	// ImmediateNextStage means that the next stage of matching is performed immediately, without waiting for the Apiserver to push.
-	ImmediateNextStage *bool `json:"immediateNextStage,omitempty"`
+	ImmediateNextStage  *bool                `json:"immediateNextStage,omitempty"`
+	ImpersonationConfig *ImpersonationConfig `json:"impersonationConfig,omitempty"`
```

Comment:
I have a feeling this might be misleading, everything in `next` will be impersonation.

Reply:
Is the concern that it could be unknowingly used for other things? or that this field should be named better to reflect it is only for the status update?

Reply:
> or that this field should be named better to reflect it is only for the status update?

Yes

Reply:
changed the field with better name.